etor/src/test/groovy/gov/hhs/cdc/trustedintermediary/etor/ruleengine/transformation/custom/MapLocalObservationCodesTest.groovy

```diff
@@ -0,0 +1,269 @@
+        def transformedCodingList = transformedObservation.getCode().getCoding()
+        transformedCodingList.size() == 1
+
+        observation.code.coding == transformedCodingList
+
+        where:
+        codingSystem << ["coding", "alt-coding"]
+    }
+
+    def "When no observation identifier, the observation does not change"() {
+        given:
+        def bundle = HapiFhirHelper.createMessageBundle(messageTypeCode: 'ORU_R01')
+
+        // Add an observation with an observation value and a status, but no observation identifier
+        def observation = new Observation()
+        observation.status = Observation.ObservationStatus.FINAL
+        def valueCoding = new Coding()
+        valueCoding.code = "123456"
+        observation.valueCodeableConcept.coding.add(valueCoding)
+        bundle.addEntry(new Bundle.BundleEntryComponent().setResource(observation))
+
+        when:
+        transformClass.transform(new HapiFhirResource(bundle), null)
+
+        then:
+        def transformedObservation = HapiHelper.resourceInBundle(bundle, Observation.class)
+
+        observation == transformedObservation
+    }
+
+    def "When message has multiple observations, local and non-local codes are handled appropriately"() {
+        given:
+        final String FHIR_ORU_PATH = "../CA/020_CA_ORU_R01_CDPH_OBX_to_LOINC_1_hl7_translation.fhir"
+        def fhirResource = ExamplesHelper.getExampleFhirResource(FHIR_ORU_PATH)
+        def bundle = fhirResource.getUnderlyingResource() as Bundle
+        def initialObservations = HapiHelper.resourcesInBundle(bundle, Observation.class).toList()
+
+        expect:
+        initialObservations.size() == 114
+
+        when:
+        transformClass.transform(new HapiFhirResource(bundle), null)
+
+        then:
+        def transformedObservations = HapiHelper.resourcesInBundle(bundle, Observation.class).toList()
+        transformedObservations.size() == 114
+
+        // Assortment of LOINC codes - ensure they are left as-is
+        def initialLoinc1 = getObservationByCode(initialObservations, "57721-3")
+        def transformedLoinc1 = getObservationByCode(transformedObservations, "57721-3")
+        initialLoinc1 == transformedLoinc1
+
+        def initialLoinc2 = getObservationByCode(initialObservations, "8339-4")
+        def transformedLoinc2 = getObservationByCode(transformedObservations, "8339-4")
+        initialLoinc2 == transformedLoinc2
+
+        def initialLoinc3 = getObservationByCode(initialObservations, "54104-5")
+        def transformedLoinc3 = getObservationByCode(transformedObservations, "54104-5")
+        initialLoinc3 == transformedLoinc3
+
+        // Mappable local code to LOINC - should have mapped code added
+        def mappedLoinc = getObservationByCode(transformedObservations, "99717-33")
+        mappedLoinc.code.coding.size() == 2
+
+        evaluateCoding(
+                mappedLoinc.code.coding[0],
+                "85268-1",
```

Comment:
I'd put these values into the enum as well

Reply:
Same as above regarding the local codes.